added few extra options

juliovedovatto opened the pull request:
Great script, solved my problem!

However, I had to change some things to achieve what I wanted.

* I added the option to extend button size.
* I added option to change the limit of speech (patience).
* I added support to see if the browser supports input elements with speech (x-webkit-speech)

Cheers!

konnng-dev committed:
added input attribute "data" options for button size and patience threshold

konnng-dev committed:
readme updated

konnng-dev committed:
improved speech native support for input elements

Daniel-Hug commented on speech-input.js:
```diff
@@ -1,14 +1,15 @@
 
 	var talkMsg = 'Speak now';
 
 	// seconds to wait for more input after last
```
This comment doesn't apply here anymore.

Daniel-Hug commented on README.md:
````diff
@@ -21,6 +21,27 @@ Yes. It uses the value of the `lang` attribute on the `<html>` element by defaul
+<input type="text" class="speech-input" data-patience="2">
+```
+
+*(The above example will give 2 seconds limit after you stop speeching)*
````
speeching > speaking :)

Daniel-Hug reviewed (changes requested):
This is great! Thank you Julio. Could you tweak these 2 things though? Also, I'm curious do you know of any modern browsers that still support x-webkit-speech? Last I checked, Chrome had not only deprecated it, but discontinued support in the most recent versions.

konnng-dev committed:
Pull request fixes

juliovedovatto commented:
Ok. Created commit 8fd20d8

* Fixed typo on README
* Improved the patience threshold code
* Removed deprecated `x-webkit-speech`.
  * I saw few articles using, plus using the `webkitSpeechRecognition` API.
  * Now I found a recent article talking about this depreciation http://becometechie.com/add-voice-recognition-website/

Please check carefully ;)

Thanks for you great work and patience.

Cheers!

Daniel-Hug commented on speech-input.js:
```diff
@@ -3,12 +3,11 @@
-
 	// seconds to wait for more input after last
-	var patience = 6;
+  var defaultPatienceThreshold = 6;
```
I was using tabs. These are spaces. Let's pick one and stick with it.

Daniel-Hug commented on speech-input.js:
```diff
@@ -25,8 +25,8 @@
+		var buttonSize = 0.8 * (inputEl.dataset.buttonsize || inputHeight);
+		var buttonTop = 0.1 * (inputEl.dataset.buttonsize || inputHeight);
 
 		// max size for textareas
```
With this setup, if you set a large `data-buttonsize` value on a `<textarea>`, it will be overridden by this "max size for textareas" JS code that follows. Is this what you're wanting?

Daniel-Hug reviewed (changes requested):
A couple more things?

konnng-dev committed:
Changed from spaces to tabs in code's indentation.

juliovedovatto commented:
Done. Commit e76314384fc49d6b724aced1f95cd4612d4f9793

* Fixed the tabs vs spaces. Now using tabs (my frontend text editor by default uses 2 spaces)
* I left the `<textarea>` condition to avoid create a big speech button. We can remove this piece of code if you think it's better to give full freedom to the user changes whenever he wants.

Daniel-Hug commented:
I do like to leave the user with options, so I merged this and added d094e9190ea5d99842c7f286fd97d033c05a4b5a which prevents `data-buttonsize` from being overridden. Thanks again.
